app/connectors/ApplicationConnector.scala

```diff
@@ -36,12 +36,22 @@ trait ApplicationConnector {
   val http: HttpPost with HttpGet
 
   def approveUplift(applicationId: String, gatekeeperUserId: String)(implicit hc: HeaderCarrier): Future[ApproveUpliftSuccessful] =
-    http.POST(s"$applicationBaseUrl/application/$applicationId/approve-uplift", ApproveUpliftRequest(gatekeeperUserId), Seq("Content-Type" -> "application/json"))
+    http.POST(s"$applicationBaseUrl/application/$applicationId/approve-uplift",
+      ApproveUpliftRequest(gatekeeperUserId), Seq("Content-Type" -> "application/json"))
       .map(_ => ApproveUpliftSuccessful)
       .recover {
-        case e: Upstream4xxResponse if (e.upstreamResponseCode == 412) => throw new ApproveUpliftPreconditionFailed
+        case e: Upstream4xxResponse if (e.upstreamResponseCode == 412) => throw new PreconditionFailed
       }
 
+  def rejectUplift(applicationId: String, gatekeeperUserId: String, rejectionReason: String)(implicit hc: HeaderCarrier) : Future[RejectUpliftSuccessful] =
+    http.POST(s"$applicationBaseUrl/application/$applicationId/reject-uplift",
+      RejectUpliftRequest(gatekeeperUserId, rejectionReason), Seq("Content-Type" -> "application/json"))
+      .map(_ => RejectUpliftSuccessful)
+      .recover {
+        case e: Upstream4xxResponse if (e.upstreamResponseCode == 412) => throw new PreconditionFailed
```

Comment:
We seem to be using Precondition Failed to prevent invalid transitions - but the rfc says it's for a different purpose: 

>  The precondition given in one or more of the request-header fields evaluated to false when it was tested on the server. This response code allows the client to place preconditions on the current resource metainformation (header field data) and thus prevent the requested method from being applied to a resource other than the one intended. 

source: https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html

I don't think we should change this - especially not as part of this pull request - just thought I should comment in case we want to use this pattern again in the future.